### ISSUE TITLE
tinyproxy 1.9.0 (2018-05-26) using a remote general upstream socks5 proxy with authentication is failing with 'no entity' message in tinyproxy.log

### ISSUE THREAD

**Issue by denshigomi** [tinyproxy/tinyproxy#172]
I have a working remote socks5 proxy that requires authentication. This curl returns the expected response:
`$ curl --socks5 'username:password@46.166.186.212:1080' https://www.google.com/`
I compiled tinyproxy with Upstream proxy support:
```
$ tinyproxy -v
tinyproxy 1.9.0
$ tinyproxy -h
...
Features compiled in:
    XTinyproxy header
    Filtering
    Transparent proxy support
    Reverse proxy support
    Upstream proxy support
...
```
I setup tinyproxy.config to forward generally to the upstream socks5 proxy:
```
User tinyproxy
Group tinyproxy
Port 8888
Timeout 600
DefaultErrorFile "/usr/share/tinyproxy/default.html"
StatFile "/usr/share/tinyproxy/stats.html"
Logfile "/var/log/tinyproxy/tinyproxy.log"
LogLevel Info
PidFile "/run/tinyproxy/tinyproxy.pid"
MaxClients 100
MinSpareServers 5
MaxSpareServers 20
StartServers 10
MaxRequestsPerChild 0
Allow 127.0.0.1
ViaProxyName "tinyproxy"
ConnectPort 443
ConnectPort 563
upstream socks5 username:password@46.166.186.212:1080
```
I start tinyproxy:
```
$ sudo /usr/local/bin/tinyproxy -c /home/username/tinyproxy.config
$ sudo cat /var/log/tinyproxy/tinyproxy.log
INFO      May 26 15:23:56 [4824]: Initializing tinyproxy ...
INFO      May 26 15:23:56 [4824]: Reloading config file
INFO      May 26 15:23:56 [4824]: Setting "Via" header to 'tinyproxy'
INFO      May 26 15:23:56 [4824]: Adding Port [443] to the list allowed by CONNECT
INFO      May 26 15:23:56 [4824]: Adding Port [563] to the list allowed by CONNECT
INFO      May 26 15:23:56 [4824]: Added upstream socks5 46.166.186.212:1080 for [default]
INFO      May 26 15:23:56 [4824]: listen_sock called with addr = '(NULL)'
INFO      May 26 15:23:56 [4824]: trying to listen on host[0.0.0.0], family[2], socktype[1], proto[6]
INFO      May 26 15:23:56 [4824]: listening on fd [0]
INFO      May 26 15:23:56 [4824]: trying to listen on host[::], family[10], socktype[1], proto[6]
INFO      May 26 15:23:56 [4824]: listening on fd [1]
INFO      May 26 15:23:56 [4824]: Now running as group "tinyproxy".
INFO      May 26 15:23:56 [4824]: Now running as user "tinyproxy".
INFO      May 26 15:23:56 [4824]: Creating child number 1 of 10 ...
INFO      May 26 15:23:56 [4824]: Creating child number 2 of 10 ...
INFO      May 26 15:23:56 [4824]: Creating child number 3 of 10 ...
INFO      May 26 15:23:56 [4824]: Creating child number 4 of 10 ...
INFO      May 26 15:23:56 [4824]: Creating child number 5 of 10 ...
INFO      May 26 15:23:56 [4824]: Creating child number 6 of 10 ...
INFO      May 26 15:23:56 [4824]: Creating child number 7 of 10 ...
INFO      May 26 15:23:56 [4824]: Creating child number 8 of 10 ...
INFO      May 26 15:23:56 [4824]: Creating child number 9 of 10 ...
INFO      May 26 15:23:56 [4824]: Creating child number 10 of 10 ...
INFO      May 26 15:23:56 [4824]: Finished creating all children.
INFO      May 26 15:23:56 [4824]: Setting the various signals.
INFO      May 26 15:23:56 [4824]: Starting main loop. Accepting connections.
```
I try to use tinyproxy, but my curl fails:
```
$ curl -v --proxy 'http://127.0.0.1:8888' https://www.google.com/
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to 127.0.0.1 (127.0.0.1) port 8888 (#0)
* allocate connect buffer!
* Establish HTTP proxy tunnel to www.google.com:443
> CONNECT www.google.com:443 HTTP/1.1
> Host: www.google.com:443
> User-Agent: curl/7.58.0
> Proxy-Connection: Keep-Alive
> 
* Proxy CONNECT aborted
* CONNECT phase completed!
* Connection #0 to host 127.0.0.1 left intact
curl: (56) Proxy CONNECT aborted
```
tinyproxy.log show a 'no entity' message, but little else:
```
$ sudo cat /var/log/tinyproxy/tinyproxy.log
CONNECT   May 26 15:29:58 [4831]: Connect (file descriptor 7): localhost [127.0.0.1]
CONNECT   May 26 15:29:58 [4831]: Request (file descriptor 7): CONNECT www.google.com:443 HTTP/1.1
INFO      May 26 15:29:58 [4831]: Found upstream proxy socks5 46.166.186.212:1080 for www.google.com
INFO      May 26 15:29:58 [4831]: opensock: opening connection to 46.166.186.212:1080
INFO      May 26 15:29:58 [4831]: opensock: getaddrinfo returned for 46.166.186.212:1080
CONNECT   May 26 15:29:58 [4831]: Established connection to socks5 proxy "46.166.186.212" using file descriptor 8.
INFO      May 26 15:29:59 [4831]: no entity
```
If I remove the upstream proxy from tinyproxy.config, the curl works. Is something wrong with upstream socks5 authentication? Am I doing something wrong?

**Comment by rofl0r** [comment]
what's the program used by the upstream socks5 proxy ? can you check with wireshark what happens ?

**Comment by denshigomi** [comment]
I don't know what program the upstream socks5 proxy uses.
I've been able to get other local socks5 servers (not tinyproxy) to forward to the upstream socks5 server using auth method 0x02 username/password.
Attached to this post is a wireshark dump of what happens when I use tinyproxy to forward to the upstream socks5 proxy.
The username and password have been changed for security.
[tinyproxy-wireshark-no-entity.zip](https://github.com/tinyproxy/tinyproxy/files/2045742/tinyproxy-wireshark-no-entity.zip)
